docs/index.adoc

```diff
@@ -3620,6 +3620,9 @@ fun main(args: Array<String>) {
 
 CAUTION: Older versions of picocli had some limited exit code support where picocli would call `System.exit`, but this is now deprecated.
 
+When an application is not appropriate to call `System.exit`, for example it runs on the caller's process, you can take a different action based on the returned `int`.
```

Comment:
This is the introduction section of the manual, I want to keep it very simple here, so I don't want to mention alternatives to calling `System.exit` just yet.

Let's remove the first sentence and only keep the following sentence that links to `<<Usage of System.exit>>`